src/main/java/io/pravega/connectors/flink/FlinkPravegaTableFactoryBase.java

```diff
@@ -0,0 +1,509 @@
+        properties.add(CONNECTOR_CONNECTION_CONFIG_SECURITY_AUTH_TYPE);
+        properties.add(CONNECTOR_CONNECTION_CONFIG_SECURITY_AUTH_TOKEN);
+        properties.add(CONNECTOR_CONNECTION_CONFIG_SECURITY_VALIDATE_HOSTNAME);
+        properties.add(CONNECTOR_CONNECTION_CONFIG_SECURITY_TRUST_STORE);
+
+        properties.add(CONNECTOR_READER);
+        properties.add(CONNECTOR_READER_STREAM_INFO);
+        properties.add(CONNECTOR_READER_STREAM_INFO + ".#." + CONNECTOR_READER_STREAM_INFO_SCOPE);
+        properties.add(CONNECTOR_READER_STREAM_INFO + ".#." + CONNECTOR_READER_STREAM_INFO_STREAM);
+        properties.add(CONNECTOR_READER_STREAM_INFO + ".#." + CONNECTOR_READER_STREAM_INFO_START_STREAMCUT);
+        properties.add(CONNECTOR_READER_STREAM_INFO + ".#." + CONNECTOR_READER_STREAM_INFO_END_STREAMCUT);
+        properties.add(CONNECTOR_READER_READER_GROUP);
+        properties.add(CONNECTOR_READER_READER_GROUP_UID);
+        properties.add(CONNECTOR_READER_READER_GROUP_SCOPE);
+        properties.add(CONNECTOR_READER_READER_GROUP_NAME);
+        properties.add(CONNECTOR_READER_READER_GROUP_REFRESH_INTERVAL);
+        properties.add(CONNECTOR_READER_READER_GROUP_EVENT_READ_TIMEOUT_INTERVAL);
+        properties.add(CONNECTOR_READER_READER_GROUP_CHECKPOINT_INITIATE_TIMEOUT_INTERVAL);
+
+        properties.add(CONNECTOR_WRITER);
+        properties.add(CONNECTOR_WRITER_SCOPE);
+        properties.add(CONNECTOR_WRITER_STREAM);
+        properties.add(CONNECTOR_WRITER_MODE);
+        properties.add(CONNECTOR_WRITER_TXN_LEASE_RENEWAL_INTERVAL);
+        properties.add(CONNECTOR_WRITER_ROUTING_KEY_FILED_NAME);
+
+        // schema
+        properties.add(SCHEMA() + ".#." + SCHEMA_TYPE());
+        properties.add(SCHEMA() + ".#." + SCHEMA_NAME());
+        properties.add(SCHEMA() + ".#." + SCHEMA_FROM());
+
+        // time attributes
+        properties.add(SCHEMA() + ".#." + SCHEMA_PROCTIME());
+        properties.add(SCHEMA() + ".#." + ROWTIME_TIMESTAMPS_TYPE());
+        properties.add(SCHEMA() + ".#." + ROWTIME_TIMESTAMPS_FROM());
+        properties.add(SCHEMA() + ".#." + ROWTIME_TIMESTAMPS_CLASS());
+        properties.add(SCHEMA() + ".#." + ROWTIME_TIMESTAMPS_SERIALIZED());
+        properties.add(SCHEMA() + ".#." + ROWTIME_WATERMARKS_TYPE());
+        properties.add(SCHEMA() + ".#." + ROWTIME_WATERMARKS_CLASS());
+        properties.add(SCHEMA() + ".#." + ROWTIME_WATERMARKS_SERIALIZED());
+        properties.add(SCHEMA() + ".#." + ROWTIME_WATERMARKS_DELAY());
+
+        // format wildcard
+        properties.add(FORMAT() + ".*");
+
+        return properties;
+    }
+
+    protected abstract String getVersion();
+
+    protected DescriptorProperties getValidatedProperties(Map<String, String> properties) {
+        final DescriptorProperties descriptorProperties = new DescriptorProperties(true);
+        descriptorProperties.putProperties(properties);
+
+        new SchemaValidator(true, false, false).validate(descriptorProperties);
```

Comment:
These bool arguments are hard to interpret, maybe a line of comment before that line could clarify the actual options we pass to the `SchemaValidator` object.